src/System.Console/src/System/IO/SyncTextReader.cs

```diff
@@ -11,7 +11,6 @@ namespace System.IO
     internal sealed class SyncTextReader : TextReader
     {
         private readonly TextReader _in;
-        private readonly object _methodLock = new object();
```

Comment:
Can you add a comment to at least one of the `lock(this)` that it is intentional so nobody tries the same change in the future? Or is there a way to assert this in a unit test?